sopel/modules/reddit.py

```diff
@@ -266,14 +312,19 @@ def subreddit_info(bot, trigger, match, commanded=False):
 
     try:
         s = r.subreddit(match)
-        s.subreddit_type
+        if match_lower not in ['all', 'popular']:
+            s.subreddit_type
     except prawcore.exceptions.Forbidden:
         bot.say("r/" + match + " appears to be a private subreddit!")
         return NOLIMIT
     except prawcore.exceptions.NotFound:
         bot.say("r/" + match + " appears to be a banned subreddit!")
         return NOLIMIT
 
+    if sorting in ['hot', 'new', 'top', 'controversial', 'gilded', 'rising', 'sticky', 'random']:
+        subreddit_sorting(bot, trigger, s, sorting)
+        return NOLIMIT
```

Comment:
This effectively disables (the non-existent) rate-limiting on fetching subreddit info. Let's write it with the future possibility of adding a rate limit in mind:

```suggestion
        return subreddit_sorting(bot, trigger, s, sorting)
```

`subreddit_sorting()` returns `NOLIMIT` on its own if appropriate, AFAICT.